ra/saphana-filesystem-lib

```diff
@@ -0,0 +1,438 @@
+    #
+    # global
+    #
+    NODENAME=$(crm_node -n)
+    SID="$OCF_RESKEY_SID"
+    InstanceNr="$OCF_RESKEY_InstanceNumber"
+    InstanceName="HDB${InstanceNr}"
+    fs_hana_shared="${OCF_RESKEY_DIRECTORY:-/hana/shared/$SID}"
+    HANA_CALL_TIMEOUT="${OCF_RESKEY_HANA_CALL_TIMEOUT:-90}"
+    td_hana_shared="$fs_hana_shared/.suse_SAPHanaFilesystem/${NODENAME}"
+    pp_hana_shared="/dev/shm/poison_pill_${SID}"
+    sid="${SID,,}"
+    export sidadm="${sid}adm"
+    #
+    # init attribute definitions
+    #
+    saphana_init_attribute_definitions
+    SAPHanaFilter=$(get_hana_attribute "X" "${ATTR_NAME_HANA_FILTER[@]}")
+    mapfile -t otherNodes < <( cluster_get_other_nodes "${NODENAME}" ) # the syntax < <(..) is intended
+    super_ocf_log info "FLOW ${FUNCNAME[0]} rc=$OCF_SUCCESS"
+    return "$OCF_SUCCESS"
+} # end function shfs_init
+
+#
+#############################################################################
+#
+# function: shfs_start - start a hana pseudo FS resource
+# params:   -
+# globals:  OCF_*
+# shfs_start : Start the SAP HANA pseudo FS resource instance
+# TODO: should shfs_start trigger an start error, if the FS is not usable at this point of time (running an internal shfs_monitor)
+#
+function shfs_start() {
+    super_ocf_log info "FLOW ${FUNCNAME[0]} ()"
+    local rc="$OCF_NOT_RUNNING"
+    local ha_ps_rc
+    shfs_reset_poison_pill
+    ha_pseudo_resource "$OCF_RESOURCE_INSTANCE" start; ha_ps_rc="$?"
+    case "$ha_ps_rc" in
+        0 ) rc="$OCF_SUCCESS";;
+        * ) rc="$OCF_NOT_RUNNING";;
+    esac
+    super_ocf_log info "FLOW ${FUNCNAME[0]} rc=$rc"
+    return "$rc"
+} # end function shfs_start
+
+#
+# function: shfs_stop - stop a hana pseudo FS resource 
+# params:   -
+# globals:  OCF_*(r), SAPCONTROL(r), SID(r), InstanceName(r)
+# shfs_stop: Stop the SAP HANA pseudo FS resource instance
+#
+function shfs_stop() {
+    super_ocf_log info "FLOW ${FUNCNAME[0]} ()"
+    local rc=0
+    local ha_ps_rc
+    ha_pseudo_resource "$OCF_RESOURCE_INSTANCE" stop; ha_ps_rc="$?"
+    if [ -e "$pp_hana_shared" ]; then
+	rm "$pp_hana_shared"
+        super_ocf_log info "RA poison pill detected - reporting stop error - sleep 10s"
+	sleep 10
+        rc="$OCF_ERR_GENERIC"
+    else
+	rc="$ha_ps_rc"
+    fi
+    super_ocf_log info "FLOW ${FUNCNAME[0]} rc=$rc"
+    return "$rc"
+} # end function shfs_stop
+
+function shfs_reset_poison_pill() {
+    if [ -e "$pp_hana_shared" ]; then
+        super_ocf_log info "reset old RA poison pill"
+	rm "$pp_hana_shared"
+    fi
+}
+
+function shfs_error_simulation() {
+    if [ -e "/tmp/break_SAPHanaFilesystem_${SID}" ]; then
+        rm "/tmp/break_SAPHanaFilesystem_${SID}"
+        return 0
+    else
+        return 1
+    fi 
+}
+
+function shfs_test_directory()
+{
+    local rc1=0 rc2=0 rc3=0
```

Comment:
rc3 is defines as local but not used in the function.
Can you please remove the variable

Reply:
Yes we changed the test and removed a needed 3rd return code in the code but have left-over the variable in the definition. should be solved in commit 272f2d1